distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Enso_Path.enso

```diff
@@ -22,7 +23,9 @@ type Enso_Path
     ## PRIVATE
     parse (path : Text) -> Enso_Path =
         if path.starts_with Enso_Path.protocol_prefix . not then Error.throw (Illegal_Argument.Error "Invalid path - it should start with `enso://`.") else
-            raw_segments = path.drop Enso_Path.protocol_prefix.length . split Enso_Path.delimiter
+            raw_segments = if path.match ".*%\d\d.*" . not then path.drop Enso_Path.protocol_prefix.length . split Enso_Path.delimiter else
+                uri = URI.parse.path
+                [if path.starts_with "enso://~" then "~" else uri.host] + (uri.path.split Enso_Path.delimiter)
```

Comment:
I appreciate wanting to handle URL encoded paths, it is probably right.

Note however that currently in the cloud nothing stopped me from creating a directory called **literally** `a%26b`. I.e. the title **contains** the `%` sign.

This change will make such directories inaccessible, because the `%26` will get url-decoded, but it is **literally** in the directory name.

I think this changes require introducing stricter requirements on asset names on the Cloud side.

Reply:
Why is the space causing the URL cannot be copied? Not sure if I understand (but I don't want to undermine the problem, just trying to understand the cause).

Sure if this solution moves as forward that is good, just noting that it can very easily lead to breakage. I think we should disallow some classes of characters from usernames and asset names in the Cloud to avoid issues like that. The sooner we do that the better as users can already be creating assets that would be forbidden afterwards and that may cause compatibility problems.

Reply:
I see.

Ah I kind of forgot that an URL containing `%26` can still be URLencoded and then decoded, so the user just needs to paste `enso://Users/name/a%2526b` so this is a smaller issue than I originally thought.

Reply:
So it's probably okay, but I think we still need to think about restrictions on naming in the Cloud, especially in the context of the other Java `Path` issue.

Reply:
Will move to a regex reproduction of glob in a separate PR

Reply:
https://github.com/enso-org/enso/issues/12562